sphinx/builders/gettext.py

```diff
@@ -237,7 +237,8 @@ def _collect_templates(self):
 
     def _extract_from_template(self):
         # type: () -> None
-        files = self._collect_templates()
+        files = list(self._collect_templates())
+        files.sort()
```

Comment:
Note: I think this is responsibility of `_collect_templates()`. On the other hand, a compatibility is another topic of maintenance. So I'd like to go with as is.